src/physics.cpp

```diff
@@ -157,7 +157,12 @@ void sample_neutron_reaction(Particle& p)
 
   // Play russian roulette if survival biasing is turned on
   if (settings::survival_biasing) {
-    if (p.wgt() < settings::weight_cutoff) {
+    double weight_cutoff = settings::weight_cutoff;
+    // if survival normalization is applicable, use normalized weight cutoff
+    if((settings::source_file || settings::surf_source_read)&&(settings::survival_normalization)){
+      weight_cutoff  = p.wgt_cutoff();
+    }
+    if (p.wgt() < weight_cutoff) {
```

Comment:
If we only add the source weight to the particle object,I think this section could be simplified to just `russian_roulette(p, settings::weight_survive*p.wgt0()`